serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/primitive/PrimitiveObjectInspectorUtils.java

```diff
@@ -384,7 +384,7 @@ public static PrimitiveTypeEntry getTypeEntryFromPrimitiveWritableClass(
    * Get the TypeEntry for the given base type name (int, varchar, etc).
    */
   public static PrimitiveTypeEntry getTypeEntryFromTypeName(String typeName) {
-    return typeNameToTypeEntry.get(typeName);
+    return typeNameToTypeEntry.get(typeName.toLowerCase());
```

Comment:
> The changes proposed in this pull request aim to make the SerDe API case-insensitive for the data type "string".
This change seems to affect all primitive type and not only "string" type.
Can this cause any problems?

Reply:
@difin Thanks for the review
Yes, adding return typeNameToTypeEntry.get(typeName.toLowerCase()); is fine. This ensures that the type name is always treated in lowercase, which is consistent with how the types are registered in the typeNameToTypeEntry map.
<img width="1049" alt="image" src="https://github.com/user-attachments/assets/a2f7fd28-b3b6-4e32-a7ba-80b1b223b642">